cluster/gce/trusty/node.yaml

```diff
@@ -148,59 +150,83 @@ post-stop exec sleep 10
 	. /etc/kube-env
-	ARGS="--v=2"
-	if [ -n "${KUBEPROXY_TEST_ARGS:-}" ]; then
-		ARGS="${KUBEPROXY_TEST_ARGS}"
-	fi
-	BINARY_PATH="/usr/bin/kube-proxy"
-	if [ "${TEST_CLUSTER:-}" = "true" ]; then
-		BINARY_PATH="/usr/local/bin/kube-proxy"
-	fi
-	${BINARY_PATH} \
-		--master=https://${KUBERNETES_MASTER_NAME} \
-		--kubeconfig=/var/lib/kube-proxy/kubeconfig \
-		${ARGS}
+	restart_docker_daemon
 end script
 
-# Wait for 10s to start kube-proxy again.
-post-stop exec sleep 10
-
 --====================================
 MIME-Version: 1.0
 Content-Type: text/upstart-job; charset="us-ascii"
 Content-Transfer-Encoding: 7bit
-Content-Disposition: attachment; filename="kube-docker.conf"
+Content-Disposition: attachment; filename="kube-proxy.conf"
 
 #upstart-job
 
-description "Restart docker daemon"
+description "Start kube-proxy static pod"
 
-# The condition "stopped kube-install-additional-packages" is to avoid
-# breaking nsenter installation, which is through a docker container.
-# It can be removed if we find a better way to install nsenter.
-start on started kubelet and stopped kube-install-additional-packages
+start on stopped kube-docker
 
 script
 	set -o errexit
 	set -o nounset
 
 	. /etc/kube-configure.sh
-	restart_docker_daemon
+	. /etc/kube-env
+	prepare_log_file "/var/log/kube-proxy.log"
+	# Load the docker image from file /run/kube-docker-files/kube-proxy.tar.
+	for i in {1..3}; do
+		echo "Try to load docker image file kube-proxy.tar"
+		timeout 30 docker load -i /run/kube-docker-files/kube-proxy.tar 1>/dev/null 2>&1
+		if [ "$?" = "0" ]; then
+			break
+		fi
+		sleep 5
+	done
+	# Copy the manifest to /tmp to manipulate
+	tmp_file="/tmp/kube-proxy.manifest"
+	cp -f /run/kube-manifests/kubernetes/kube-proxy.manifest ${tmp_file}
+	# Remove the lines of salt configuration and replace variables with values.
+	# NOTE: Changes to variable names in cluster/saltbase/salt/kube-proxy/kube-proxy.manifest
+	# may break this upstart job.
+	sed -i "/^ *{%/d" ${tmp_file}
+	kubeconfig="--kubeconfig=\/var\/lib\/kube-proxy\/kubeconfig"
+	kube_docker_registry="gcr.io\/google_containers"
+	if [ -n "${KUBE_DOCKER_REGISTRY:-}" ]; then
+		kube_docker_registry=${KUBE_DOCKER_REGISTRY}
+	fi
+	kube_proxy_docker_tag=$(cat /run/kube-docker-files/kube-proxy.docker_tag)
+	test_args=""
+	log_level="--v=2"
+	if [ -n "${KUBEPROXY_TEST_ARGS:-}" ]; then
+		test_args="${KUBEPROXY_TEST_ARGS}"
+		# test_args should already contain log level setting.
+		log_level=""
+	fi
+	api_servers="--master=https:\/\/${KUBERNETES_MASTER_NAME}"
+	sed -i -e "s/{{kubeconfig}}/${kubeconfig}/g" ${tmp_file}
+	sed -i -e "s/{{pillar\['kube_docker_registry'\]}}/${kube_docker_registry}/g" ${tmp_file}
+	sed -i -e "s/{{pillar\['kube-proxy_docker_tag'\]}}/${kube_proxy_docker_tag}/g" ${tmp_file}
+	sed -i -e "s/{{test_args}}/${test_args}/g" ${tmp_file}
+	sed -i -e "s/{{log_level}}/${log_level}/g" ${tmp_file}
+	sed -i -e "s/{{api_servers_with_port}}/${api_servers}/g" ${tmp_file}
+
```

Comment:
@yifan-gu I had thought about which way to go, but I got an opposite conclusion to yours. I think making a copy in cluster/gce/kube-manifests will make it more difficult to maintain. First of all, either way has maintenance problems, as long as salt is still widely used. In the way of this PR, it is broken when the salt file has variable related changes, e.g., adding a new variable, removing an old one, or renaming a variable, because the shell code is for replacing variables with values. But it is not affected by many other changes, including any fixed-text changes to the config, spec, metadata, or command line. On the other hand, if making a copy in cluster/gce/kube-manifests, it will be affected by any change in the salt file. Whenever there is a change to the salt file, we have to update the copy in cluster/gce/kube-manifests accordingly to have a consistent version. So, I try to avoid making a copy.